testsuite/elytron/src/test/java/org/wildfly/test/security/common/AbstractSystemPropertiesServerSetupTask.java

```diff
@@ -0,0 +1,176 @@
+        if (LOGGER.isInfoEnabled()) {
+            LOGGER.trace("Adding system properties.");
+        }
+        systemProperties = getSystemProperties();
+        if (systemProperties == null || systemProperties.length == 0) {
+            LOGGER.warn("No system property configured in the ServerSetupTask");
+            return;
+        }
+
+        final List<ModelNode> updates = new ArrayList<ModelNode>();
+
+        for (SystemProperty systemProperty : systemProperties) {
+            final String propertyName = systemProperty.getName();
+            if (propertyName == null || propertyName.trim().length() == 0) {
+                LOGGER.warn("Empty property name provided.");
+                continue;
+            }
+            ModelNode op = new ModelNode();
+            op.get(OP).set(ADD);
+            op.get(OP_ADDR).add(SYSTEM_PROPERTY, propertyName);
+            op.get(ModelDescriptionConstants.VALUE).set(systemProperty.getValue());
+            updates.add(op);
+        }
+        CoreUtils.applyUpdates(updates, managementClient.getControllerClient());
+    }
+
+    /**
+     *
+     * @param managementClient
+     * @param containerId
+     * @see org.jboss.as.test.integration.security.common.AbstractSecurityDomainSetup#tearDown(org.jboss.as.arquillian.container.ManagementClient,
+     *      java.lang.String)
+     */
+    public final void tearDown(ManagementClient managementClient) throws Exception {
+        if (LOGGER.isInfoEnabled()) {
+            LOGGER.trace("Removing system properties.");
+        }
+        if (systemProperties == null || systemProperties.length == 0) {
+            return;
+        }
+
+        final List<ModelNode> updates = new ArrayList<ModelNode>();
+
+        for (SystemProperty systemProperty : systemProperties) {
+            final String propertyName = systemProperty.getName();
+            if (propertyName == null || propertyName.trim().length() == 0) {
+                continue;
+            }
+            ModelNode op = new ModelNode();
+            op.get(OP).set(REMOVE);
+            op.get(OP_ADDR).add(SYSTEM_PROPERTY, propertyName);
+            updates.add(op);
+        }
+        CoreUtils.applyUpdates(updates, managementClient.getControllerClient());
+
+    }
+
+    public static SystemProperty[] mapToSystemProperties(Map<String, String> map) {
+        if (map == null || map.isEmpty()) {
+            return null;
+        }
+        final List<SystemProperty> list = new ArrayList<SystemProperty>();
+        for (Map.Entry<String, String> property : map.entrySet()) {
+            list.add(new DefaultSystemProperty(property.getKey(), property.getValue()));
+        }
+        return list.toArray(new SystemProperty[list.size()]);
+    }
+
+    // Protected methods -----------------------------------------------------
+
+    /**
+     * Returns configuration of the login modules.
+     *
+     * @return
+     */
+    protected abstract SystemProperty[] getSystemProperties();
```

Comment:
It's a copy from WildFly full. Maybe I was thinking about wider usage in the ancient times I've implemented it. I'll change it to simple Map here.